### ISSUE TITLE
[Auto Sequence DNA] v1.0.0.5 - CrashLoop when trying to find a Larvae

### ISSUE THREAD

**Issue by masteredward** [akarnokd/ThePlanetCrafterMods#41]
Hi akarnokd,

I'm facing a CrashLoop with the Incubator trying to search for a Larvae. This began to happen after crafting T2 Mutagens and after Fish terraforming phase. I have all your other mods enabled was well except the SpaceCow.

Important notes to try to reproduce the issue:

- I'm using Drones to transport Fertilizer and Mutagen from Auto-Crafter to Containers
- I'm using custom Container names
- The Inventory Stack mod is set to 999
- I have a thousands of Larvae stored in a single Container since I'm using the Teleport to Nearest Mineable auto-mine enabled. Note the issue happens independent if the the ToggleAutomatic is enabled or no
- I tried to empty all the Containers to see if the issue is gone but is still happening
- The incubators are empty
- If i disable only the Incubator this stops from happening

This is the error log:
```
[Info   :(Cheat) Auto Sequence DNA] Begin<Incubators>
[Info   :(Cheat) Auto Sequence DNA]   Container discovery
[Info   :(Cheat) Auto Sequence DNA]     Larvae <- 6000, Container2, "RocketInsects1", 
[Info   :(Cheat) Auto Sequence DNA]     Butterfly <- 6000, Container2, "RocketInsects1", 
[Info   :(Cheat) Auto Sequence DNA]     Bee <- 6000, Container2, "RocketInsects1", 
[Info   :(Cheat) Auto Sequence DNA]     Silk <- 6000, Container2, "RocketInsects1", 
[Info   :(Cheat) Auto Sequence DNA]     Fertilizer <- 205340981, Container2, "Fertilizer", (659.6, 25.5, 567.0)
[Info   :(Cheat) Auto Sequence DNA]     Phytoplankton <- 204408906, Container2, "Fish", (655.4, 25.5, 563.2)
[Info   :(Cheat) Auto Sequence DNA]     Fish <- 204408906, Container2, "Fish", (655.4, 25.5, 563.2)
[Info   :(Cheat) Auto Sequence DNA]     Mutagen <- 203816152, Container2, "Mutagen", (639.4, 25.5, 567.0)
[Info   :(Cheat) Auto Sequence DNA]     Larvae <- 201467293, Container2, "Insect", (643.6, 25.5, 567.0)
[Info   :(Cheat) Auto Sequence DNA]     Butterfly <- 201467293, Container2, "Insect", (643.6, 25.5, 567.0)
[Info   :(Cheat) Auto Sequence DNA]     Bee <- 201467293, Container2, "Insect", (643.6, 25.5, 567.0)
[Info   :(Cheat) Auto Sequence DNA]     Silk <- 201467293, Container2, "Insect", (643.6, 25.5, 567.0)
[Info   :(Cheat) Auto Sequence DNA]     Incubator <- 201974312, Incubator1, (623.5, 25.5, 581.0)
[Info   :(Cheat) Auto Sequence DNA]     Incubator <- 203149275, Incubator1, (627.5, 25.5, 581.0)
[Info   :(Cheat) Auto Sequence DNA]     Incubator <- 202120224, Incubator1, (627.5, 25.5, 565.5)
[Info   :(Cheat) Auto Sequence DNA]     Incubator <- 202871066, Incubator1, (623.5, 25.5, 565.5)
[Info   :(Cheat) Auto Sequence DNA]   Incubator: 201974312, Incubator1, (623.5, 25.5, 581.0)
[Info   :(Cheat) Auto Sequence DNA]     Picking Recipe
[Info   :(Cheat) Auto Sequence DNA]     Candidate pool:
[Info   :(Cheat) Auto Sequence DNA]       Bee1Larvae ("Bee larva") @ Chance = 100 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly1Larvae ("Butterfly Azurae larva") @ Chance = 100 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly2Larvae ("Butterfly Leani larva") @ Chance = 75 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly3Larvae ("Butterfly Fensea larva") @ Chance = 50 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly4Larvae ("Butterfly Galaxe larva") @ Chance = 75 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly6Larvae ("Butterfly Empalio larva") @ Chance = 100 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly7Larvae ("Butterfly Penga larva") @ Chance = 70 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly8Larvae ("Butterfly Chevrone larva") @ Chance = 60 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly9Larvae ("Butterfly Aemel larva") @ Chance = 50 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly5Larvae ("Butterfly Abstreus larva") @ Chance = 20 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly10Larvae ("Butterfly Liux larva") @ Chance = 50 %
[Info   :(Cheat) Auto Sequence DNA]       SilkWorm ("Silk worm") @ Chance = 70 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly16Larvae ("Butterfly Imeo larva") @ Chance = 80 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly17Larvae ("Butterfly Serena larva") @ Chance = 70 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly18Larvae ("Butterfly Golden larva") @ Chance = 15 %
[Info   :(Cheat) Auto Sequence DNA]       Fish2Eggs ("Fish Vilnus eggs") @ Chance = 50 %
[Info   :(Cheat) Auto Sequence DNA]       Fish4Eggs ("Fish Khrom eggs") @ Chance = 50 %
[Info   :(Cheat) Auto Sequence DNA]       Fish7Eggs ("Fish Tegede eggs") @ Chance = 20 %
[Info   :(Cheat) Auto Sequence DNA]       Fish8Eggs ("Fish Ecaru eggs") @ Chance = 20 %
[Info   :(Cheat) Auto Sequence DNA]       Fish9Eggs ("Fish Buyu eggs") @ Chance = 20 %
[Info   :(Cheat) Auto Sequence DNA]       Fish10Eggs ("Fish Tiloo eggs") @ Chance = 20 %
[Info   :(Cheat) Auto Sequence DNA]       Fish11Eggs ("Golden fish eggs") @ Chance = 20 %
[Info   :(Cheat) Auto Sequence DNA]     Picked: Fish8Eggs ("Fish Ecaru eggs") @ Chance = 20 %
[Info   :(Cheat) Auto Sequence DNA]       Checking inventory for ingredients
[Info   :(Cheat) Auto Sequence DNA]         Not in inventory: Phytoplankton2
[Info   :(Cheat) Auto Sequence DNA]         Not in inventory: Mutagen3
[Info   :(Cheat) Auto Sequence DNA]         Not in inventory: Fertilizer1
[Info   :(Cheat) Auto Sequence DNA]       Checking containers for missing ingredients
[Info   :(Cheat) Auto Sequence DNA]         Checking sources for ingredient category: Phytoplankton for Phytoplankton2
[Info   :(Cheat) Auto Sequence DNA]           Searching
[Info   :(Cheat) Auto Sequence DNA]         Checking sources for ingredient category: Mutagen for Mutagen3
[Info   :(Cheat) Auto Sequence DNA]           Searching
[Info   :(Cheat) Auto Sequence DNA]         Checking sources for ingredient category: Fertilizer for Fertilizer1
[Info   :(Cheat) Auto Sequence DNA]           Searching
[Info   :(Cheat) Auto Sequence DNA]         Looking for ingredient in sources: Phytoplankton2 (Phytoplankton)
[Info   :(Cheat) Auto Sequence DNA]         No source for ingredient Phytoplankton2
[Info   :(Cheat) Auto Sequence DNA]         Looking for ingredient in sources: Mutagen3 (Mutagen)
[Info   :(Cheat) Auto Sequence DNA]         Ingredient Found Mutagen3
[Info   :(Cheat) Auto Sequence DNA]         Looking for ingredient in sources: Fertilizer1 (Fertilizer)
[Info   :(Cheat) Auto Sequence DNA]         Ingredient Found Fertilizer1
[Info   :(Cheat) Auto Sequence DNA]       Recipe check: Found = 2, Required = 3
[Info   :(Cheat) Auto Sequence DNA]       Sequencing: Ingredients still missing
[Info   :(Cheat) Auto Sequence DNA]   Incubator: 203149275, Incubator1, (627.5, 25.5, 581.0)
[Info   :(Cheat) Auto Sequence DNA]     Picking Recipe
[Info   :(Cheat) Auto Sequence DNA]     Candidate pool:
[Info   :(Cheat) Auto Sequence DNA]       Bee1Larvae ("Bee larva") @ Chance = 100 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly1Larvae ("Butterfly Azurae larva") @ Chance = 100 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly2Larvae ("Butterfly Leani larva") @ Chance = 75 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly3Larvae ("Butterfly Fensea larva") @ Chance = 50 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly4Larvae ("Butterfly Galaxe larva") @ Chance = 75 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly6Larvae ("Butterfly Empalio larva") @ Chance = 100 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly7Larvae ("Butterfly Penga larva") @ Chance = 70 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly8Larvae ("Butterfly Chevrone larva") @ Chance = 60 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly9Larvae ("Butterfly Aemel larva") @ Chance = 50 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly5Larvae ("Butterfly Abstreus larva") @ Chance = 20 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly10Larvae ("Butterfly Liux larva") @ Chance = 50 %
[Info   :(Cheat) Auto Sequence DNA]       SilkWorm ("Silk worm") @ Chance = 70 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly16Larvae ("Butterfly Imeo larva") @ Chance = 80 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly17Larvae ("Butterfly Serena larva") @ Chance = 70 %
[Info   :(Cheat) Auto Sequence DNA]       Butterfly18Larvae ("Butterfly Golden larva") @ Chance = 15 %
[Info   :(Cheat) Auto Sequence DNA]       Fish2Eggs ("Fish Vilnus eggs") @ Chance = 50 %
[Info   :(Cheat) Auto Sequence DNA]       Fish4Eggs ("Fish Khrom eggs") @ Chance = 50 %
[Info   :(Cheat) Auto Sequence DNA]       Fish7Eggs ("Fish Tegede eggs") @ Chance = 20 %
[Info   :(Cheat) Auto Sequence DNA]       Fish8Eggs ("Fish Ecaru eggs") @ Chance = 20 %
[Info   :(Cheat) Auto Sequence DNA]       Fish9Eggs ("Fish Buyu eggs") @ Chance = 20 %
[Info   :(Cheat) Auto Sequence DNA]       Fish10Eggs ("Fish Tiloo eggs") @ Chance = 20 %
[Info   :(Cheat) Auto Sequence DNA]       Fish11Eggs ("Golden fish eggs") @ Chance = 20 %
[Info   :(Cheat) Auto Sequence DNA]     Picked: Butterfly6Larvae ("Butterfly Empalio larva") @ Chance = 100 %
[Info   :(Cheat) Auto Sequence DNA]       Checking inventory for ingredients
[Info   :(Cheat) Auto Sequence DNA]         Not in inventory: LarvaeBase1
[Info   :(Cheat) Auto Sequence DNA]         Not in inventory: Mutagen1
[Info   :(Cheat) Auto Sequence DNA]         Not in inventory: Fertilizer1
[Info   :(Cheat) Auto Sequence DNA]       Checking containers for missing ingredients
[Info   :(Cheat) Auto Sequence DNA]         Checking sources for ingredient category: Larvae for LarvaeBase1
[Info   :(Cheat) Auto Sequence DNA]           Searching
[Error  :(Cheat) Auto Sequence DNA] System.NullReferenceException: Object reference not set to an instance of an object
  at CheatAutoSequenceDNA.Plugin.FindIngredientsIn (System.String gid, System.Collections.Generic.Dictionary`2[TKey,TValue] itemCategories, System.String itemKey, System.Collections.Generic.List`1[T] result) [0x00036] in <ef7062efff024e34be197b83ffd24794>:0 
  at CheatAutoSequenceDNA.Plugin.StartNewResearch (SpaceCraft.GroupItem spawnTarget, System.Collections.Generic.List`1[T] currentItems, System.Collections.Generic.Dictionary`2[TKey,TValue] itemCategories, SpaceCraft.Inventory machineInventory, SpaceCraft.WorldObject machine) [0x0020f] in <ef7062efff024e34be197b83ffd24794>:0 
  at CheatAutoSequenceDNA.Plugin.HandleIncubators () [0x003b7] in <ef7062efff024e34be197b83ffd24794>:0 
  at CheatAutoSequenceDNA.Plugin+<SequencerCheckLoop>d__22.MoveNext () [0x000a2] in <ef7062efff024e34be197b83ffd24794>:0 
```

This is my configuration file:
```
## Settings file was created by plugin (Cheat) Auto Sequence DNA v1.0.0.5
## Plugin GUID: akarnokd.theplanetcraftermods.cheatautosequencedna

[General]

## Enable debugging with detailed logs (chatty!).
# Setting type: Boolean
# Default value: false
DebugMode = true

[Incubator]

## Should the Incubator auto sequence?
# Setting type: Boolean
# Default value: true
Enabled = true

## The name of the container(s) where to look for fertilizer.
# Setting type: String
# Default value: *Fertilizer
Fertilizer = Fertilizer

## The name of the container(s) where to look for mutagen.
# Setting type: String
# Default value: *Mutagen
Mutagen = Mutagen

## The name of the container(s) where to look for larvae (common, uncommon, rare).
# Setting type: String
# Default value: *Larvae
Larvae = Insect

## The name of the container(s) where to deposit the spawned butterflies.
# Setting type: String
# Default value: *Butterfly
Butterfly = Insect

## The name of the container(s) where to deposit the spawned bees.
# Setting type: String
# Default value: *Bee
Bee = Insect

## The name of the container(s) where to deposit the spawned silk worms.
# Setting type: String
# Default value: *Silk
Silk = Insect

## The name of the container(s) where to look for Phytoplankton.
# Setting type: String
# Default value: *Phytoplankton
Phytoplankton = Fish

## The name of the container(s) where to deposit the spawned fish.
# Setting type: String
# Default value: *Fish
Fish = Fish

[Sequencer]

## Should the Tree-sequencer auto sequence?
# Setting type: Boolean
# Default value: true
Enabled = true

## The name of the container(s) where to look for fertilizer.
# Setting type: String
# Default value: *Mutagen
Mutagen = Mutagen

## The name of the container(s) where to look for Tree Root.
# Setting type: String
# Default value: *TreeRoot
TreeRoot = Plant

## The name of the container(s) where to look for Flower Seeds (all kinds).
# Setting type: String
# Default value: *FlowerSeed
FlowerSeed = Plant

## The name of the container(s) where to deposit the spawned tree seeds.
# Setting type: String
# Default value: *TreeSeed
TreeSeed = Plant
```

**Comment by akarnokd** [comment]
You have a couple of weird named containers:

```
[Info   :(Cheat) Auto Sequence DNA]     Larvae <- 6000, Container2, "RocketInsects1", 
[Info   :(Cheat) Auto Sequence DNA]     Butterfly <- 6000, Container2, "RocketInsects1", 
[Info   :(Cheat) Auto Sequence DNA]     Bee <- 6000, Container2, "RocketInsects1", 
[Info   :(Cheat) Auto Sequence DNA]     Silk <- 6000, Container2, "RocketInsects1", 
```

Your naming scheme, `Insect`, matches the pin recipe's storage where you have the insect rocket pinned. For a quick workaround, use something else for naming insects.

**Comment by akarnokd** [comment]
Should be fixed with **v1.0.0.6**.

**Comment by masteredward** [comment]
Thanks for the update @akarnokd!

Now that you pointed out, I can see the issue! That's why you're using `*` in the default names! Makes a lot of sense now 😄 

Thanks for your awesome mods! I will try to help, at least, by testing them. My C# understanding is very basic!
